modules/serverless/pages/eventing/backstage-setup.adoc

```diff
@@ -0,0 +1,186 @@
+= Setup Backstage for Eventing
+:compat-mode!:
+// Metadata:
+:description: Setup Backstage for Eventing in {serverlessproductname}
+
+This page describes how to set up Backstage for Eventing in {serverlessproductname}, which allows you to discover your Eventing resources in Backstage.
+
+[IMPORTANT]
+====
+{serverlessproductname} Backstage integration is a Developer Preview feature only.
+
+Developer Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete.
+Red Hat does not recommend using them in production.
+These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.
+
+For more information about the support scope of Red Hat Developer Preview features, see https://access.redhat.com/support/offerings/devpreview/.
+====
+
+== Prerequisites
+
+* You have access to an {product-title} account with cluster administrator access.
+
+* Install the OpenShift CLI (`oc`).
+
+* Install the {serverlessoperatorname}.
+
+* Install the https://developers.redhat.com/rhdh/overview[Red Hat Developer Hub Operator] and create a `Backstage` instance.
+
+== Enable Backstage backend for Eventing
+
+. Configure `KnativeEventing` to create the necessary backend resources for Backstage:
++
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+
+  # Other spec fields omitted ...
+  # ...
+
+  config:
+    features:
+      backstage-backend: enabled <1>
+----
+<1> Enables Backstage backend deployment in Eventing namespace.
+
+. Apply the `KnativeEventing` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+== Create necessary tokens for Backstage backend and {serverlessproductname} communication
+
+Backstage backend will communicate with the {serverlessproductname} Backstage backend using a token. That token will be passed to the {serverlessproductname} backend and the it will use it to talk to the Kubernetes API server.
+
+. Create a `ServiceAccount`:
++
+[source,terminal]
+----
+$ oc -n knative-eventing create serviceaccount backstage-admin
+----
+
+. Create a `ClusterRole` that allows the `ServiceAccount` to list and get resources in the cluster:
++
+[source,terminal]
+----
+$ oc create clusterrole backstage-admin --verb=list,get --resource="*"
```

Comment:
this allows secret level access, can we restrict this to only what we need (ie addressables, triggers, subscriptions)

Reply:
I will just document that users need to adjust to their needs.